arborist/resource.go

```diff
@@ -0,0 +1,46 @@
+package arborist
+
+import ()
+
+// Representation of a resource (thing to which access is allowed). Resources
+// also have a hierarchical organization, where the children are meant to be
+// finer-grained levels of authorization. Resources must be uniquely identified
+// and the auth engine must maintain the collection of resource IDs.
+type Resource struct {
+	ID           string
+	service      *Service
```

Comment:
why is resource tied to one service?
A project doesn't need to tie to a particular service?

Reply:
Yeah I'm not sure why this made sense at the time. Decoupling in progress